### ISSUE TITLE
[resolved] not able to run distributed setup

### ISSUE THREAD

**Issue by xiyanxiyan10** [minio/minio#9252]
Bugs when run minio as cluster with data dir mounted from extended disk

## Expected Behavior

prepare on node has two disk

```
[root@localhost ~]# fdisk -l

Disk /dev/sda: 42.9 GB, 42949672960 bytes, 83886080 sectors
Units = sectors of 1 * 512 = 512 bytes
Sector size (logical/physical): 512 bytes / 512 bytes
I/O size (minimum/optimal): 512 bytes / 512 bytes
Disk label type: dos
Disk identifier: 0x00000fb1

   Device Boot      Start         End      Blocks   Id  System
/dev/sda1   *        2048     2099199     1048576   83  Linux
/dev/sda2         2099200    83886079    40893440   8e  Linux LVM
WARNING: fdisk GPT support is currently new, and therefore in an experimental phase. Use at your own discretion.

Disk /dev/sdb: 536.9 GB, 536870912000 bytes, 1048576000 sectors
Units = sectors of 1 * 512 = 512 bytes
Sector size (logical/physical): 512 bytes / 512 bytes
I/O size (minimum/optimal): 512 bytes / 512 bytes
Disk label type: gpt
Disk identifier: 5E32DF46-7286-4DC7-BCDD-BC71634E0865


#         Start          End    Size  Type            Name
 1         2048   1048575966    500G  Linux filesyste 

Disk /dev/mapper/centos-root: 37.6 GB, 37576769536 bytes, 73392128 sectors
Units = sectors of 1 * 512 = 512 bytes
Sector size (logical/physical): 512 bytes / 512 bytes
I/O size (minimum/optimal): 512 bytes / 512 bytes


Disk /dev/mapper/centos-swap: 4294 MB, 4294967296 bytes, 8388608 sectors
Units = sectors of 1 * 512 = 512 bytes
Sector size (logical/physical): 512 bytes / 512 bytes
I/O size (minimum/optimal): 512 bytes / 512 bytes
```

```
on every node 

- build part
fdisk /dev/sdb

- format it 
mkfs.ext4 /dev/sdb1 

- mount it 
mkdir /data
chmod 777 /data -R 
mount /dev/sdb1 /data
mkdir -p /opt/minio
chmod 777 /opt/minio -R

```

run 

```
#!/bin/bash

export MINIO_ACCESS_KEY=Minio
export MINIO_SECRET_KEY=Minio-8a495d8a-72f9-11ea-882c-acbc3298c95d

/root/minio server --address 0.0.0.0:9011 --config-dir /opt/minio \
http://10.4.3.25:9011/data \
http://10.4.3.26:9011/data \
http://10.4.3.27:9011/data \
http://10.4.3.28:9011/data \
```

```
run as root 
./minio.sh
```

## Current Behavior

```
Initializing data volume. Waiting for minimum 3 servers to come online. (elapsed 8m33s)
ERRO[0536] Disk http://10.4.3.25:9011/data is still unreachable  cause=disk not found source=[prepare-storage.go:202:printRetryMsg()]
ERRO[0536] Disk http://10.4.3.26:9011/data is still unreachable  cause=disk not found source=[prepare-storage.go:202:printRetryMsg()]
ERRO[0536] Initializing object layer failed              cause=
[01/04] http://10.4.3.25:9011/data : disk not found
[02/04] http://10.4.3.26:9011/data : disk not found
[03/04] http://10.4.3.27:9011/data : unformatted disk found
[04/04] /data : unformatted disk found source=[server-main.go:206:serverMain()]
ERRO[0541] Unable to shutdown http server                cause=timed out. some connections are still active. doing abnormal shutdown source=[server-main.go:208:serverMain()]
[root@localhost ~]# 
[root@localhost ~]# 

```
## Possible Solution

run minio as one node with data from system(sda) is ok 
run minio as cluster with data from system(sda) is ok 
run minio as one node with data from /data(sdb1 mounted ) is ok 

run minio as cluster with data from /data(sdb1 mounted ) is ok 

## Steps to Reproduce (for bugs)

1.up describes can reproduced it 

## Your Environment

* Version used (`minio version`):

```
Version: 2017-08-05T00:00:53Z
Release-Tag: RELEASE.2017-08-05T00-00-53Z
Commit-ID: aeafe668d8b6d25caac671d59e2b0f0473ce35d0
```

* Environment name and version (e.g. nginx 1.9.1):
* Server type and version:
* Operating System and version (`uname -a`):

```
Linux localhost.localdomain 3.10.0-862.el7.x86_64 #1 SMP Fri Apr 20 16:44:24 UTC 2018 x86_64 x86_64 x86_64 GNU/Linux
```

* Link to your project:

```
https://github.com/minio/minio/
```

**Comment by harshavardhana** [comment]
> Version: 2017-08-05T00:00:53Z
> Release-Tag: RELEASE.2017-08-05T00-00-53Z
> Commit-ID: aeafe668d8b6d25caac671d59e2b0f0473ce35d0

This looks like an arcane version can you use the latest release and let us know?

**Comment by xiyanxiyan10** [comment]
> > Version: 2017-08-05T00:00:53Z
> > Release-Tag: RELEASE.2017-08-05T00-00-53Z
> > Commit-ID: [aeafe66](https://github.com/minio/minio/commit/aeafe668d8b6d25caac671d59e2b0f0473ce35d0)
> 
> This looks like an arcane version can you use the latest release and let us know?

VERSION
2017-08-05T00:00:53Z
Thanks, It seems a little old , I will upgrade the latest release and try again

**Comment by xiyanxiyan10** [comment]
I complie the newest version(commi 73f9d8a636aeafc31876a767d9d75a197bd2e2ff) and run in three node as before
and then one node log as 
```
.
Waiting for all other servers to be online to format the disks.
Waiting for all other servers to be online to format the disks.
Waiting for all other servers to be online to format the disks.
Waiting for all other servers to be online to format the disks
```

other three node log as 

```
Waiting for the first server to format the disks.
Waiting for the first server to format the disks.
Waiting for the first server to format the disks.
```

I already mkfs.ext4 /dev/sdb1, why wait format.
Please help me to see this

It seems that they are waiting for easy other to format in a cycle

**Comment by xiyanxiyan10** [comment]
> > Version: 2017-08-05T00:00:53Z
> > Release-Tag: RELEASE.2017-08-05T00-00-53Z
> > Commit-ID: [aeafe66](https://github.com/minio/minio/commit/aeafe668d8b6d25caac671d59e2b0f0473ce35d0)
> 
> This looks like an arcane version can you use the latest release and let us know?

I build the latest one and error as my msg

**Comment by klauspost** [comment]
It doesn't look as if the servers can see each other.

**Comment by xiyanxiyan10** [comment]
1.It fix , unformat fixed in Commit-ID: aeafe66
2.not connect casued by the firewall

Thanks

**Comment by harshavardhana** [comment]
> 1.It fix , unformat fixed in Commit-ID: [aeafe66](https://github.com/minio/minio/commit/aeafe668d8b6d25caac671d59e2b0f0473ce35d0)
> 2.not connect casued by the firewall
> 
> Thanks

Thanks closing the issue
